graph/propertyconventions.md

```diff
@@ -0,0 +1,70 @@
+# Guidelines for the conventions of properties on graph types
+
+## Updating read-only properties
+
+### Convention
+
+It is a common call pattern for Microsoft Graph clients to `GET` an entity, make changes to a few properties, and then `PATCH` the same entity with the updated values.
+When clients do this, they often ignore that some properties are read-only or that a property may have a value that is allowed in a response, but not in a request.
+Because of this, workloads should support the practice that a request that updates a property should be considered a no-op if the value in the request matches the current value of the property.
+
+### Example
+
+For the following schema:
+```xml
+<EntityType Name="entity">
+  <Key>
+    <PropertyRef Name="id" />
+  </Key>
+  <Property Name="id" Type="Edm.String" Nullable="false" />
+  <Property Name="foo" Type="Edm.String" Nullable="false" /> <!--this property is read-only-->
```

Comment:
We should include one (or more) of the annotations to show how to correctly mark a property as read-only, otherwise we give the impression that workload authors don't need to mark their properties